src/util.cpp

```diff
@@ -227,211 +182,210 @@ void split_LU(Eigen::MatrixXd* A, Eigen::MatrixXd* L, Eigen::MatrixXd* U) {
     assert(L->cols() == m);
     if (m == 0 || n == 0)
         return;
-    cblas_dtrmm(CblasColMajor, CblasLeft, CblasLower, CblasTrans, CblasNonUnit, m, n, 1.0, L->data(), m, A->data(), m);
+    blas::trmm(Layout::ColMajor, Side::Left, Uplo::Lower, Op::Trans, Diag::NonUnit, m, n, 1.0, L->data(), m, A->data(), m);
 }
 
 // x2 -= A21 * x1
 void gemv_notrans(MatrixXd* A21, Segment* x1, Segment* x2) {
-    int m = A21->rows();
-    int n = A21->cols();
+    int64_t m = A21->rows();
+    int64_t n = A21->cols();
     assert(x1->size() == n);
     assert(x2->size() == m);
     if (n == 0 || m == 0)
         return;
-    cblas_dgemv(CblasColMajor, CblasNoTrans, m, n, -1.0, A21->data(), m, x1->data(), 1, 1.0, x2->data(), 1);
+    blas::gemv(Layout::ColMajor, Op::NoTrans, m, n, -1.0, A21->data(), m, x1->data(), 1, 1.0, x2->data(), 1);
 }
 
 // x2 -= A12^T x1
 void gemv_trans(MatrixXd* A12, Segment* x1, Segment* x2) {
-    int m = A12->rows();
-    int n = A12->cols();
+    int64_t m = A12->rows();
+    int64_t n = A12->cols();
     assert(x1->size() == m);
     assert(x2->size() == n);
     if (n == 0 || m == 0)
         return;
-    cblas_dgemv(CblasColMajor, CblasTrans, m, n, -1.0, A12->data(), m, x1->data(), 1, 1.0, x2->data(), 1);
+    blas::gemv(Layout::ColMajor, Op::Trans, m, n, -1.0, A12->data(), m, x1->data(), 1, 1.0, x2->data(), 1);
 }
 
 // x <- Q * x
 void ormqr_notrans(MatrixXd* v, VectorXd* h, Segment* x) {
-    int m = v->rows();
-    int n = v->cols();
+    int64_t m = v->rows();
+    int64_t n = v->cols();
     assert(h->size() == n);
     assert(x->size() == m);
     if (m == 0) 
         return;
-    int info = LAPACKE_dormqr(LAPACK_COL_MAJOR, 'L', 'N', m, 1, n, v->data(), m, h->data(), x->data(), m); 
+    int64_t info = lapack::ormqr(Side::Left, Op::NoTrans, m, 1, n, v->data(), m, h->data(), x->data(), m); 
     assert(info == 0);
 }
 
 // x <- Q^T * x
 void ormqr_trans(MatrixXd* v, VectorXd* h, Segment* x) {
-    int m = x->size();
+    int64_t m = x->size();
     // n = 1
-    int k = v->cols();
+    int64_t k = v->cols();
     assert(h->size() == k);
     assert(v->rows() == m);
     if (m == 0 || k == 0) 
         return;
-    int info = LAPACKE_dormqr(LAPACK_COL_MAJOR, 'L', 'T', m, 1, k, v->data(), m, h->data(), x->data(), m); 
+    int64_t info = lapack::ormqr(Side::Left, Op::Trans, m, 1, k, v->data(), m, h->data(), x->data(), m); 
     assert(info == 0);
 }
 
 // A <- (Q^/T) * A * (Q^/T)
-void ormqr(MatrixXd* v, VectorXd* h, MatrixXd* A, char side, char trans) {
-    int m = A->rows();
-    int n = A->cols();
-    int k = v->cols(); // number of reflectors
+void ormqr_spand(MatrixXd* v, VectorXd* h, MatrixXd* A, Side side, Op trans) {
+    int64_t m = A->rows();
+    int64_t n = A->cols();
+    int64_t k = v->cols(); // number of reflectors
     assert(h->size() == k);
     if (m == 0 || n == 0)
         return;
-    if(side == 'L') // Q * A or Q^T * A
+    if (side == Side::Left) // Q * A or Q^T * A
```

Comment:
Probably would be best to have braces here, but since they were missing originally, it's OKay :)